### PR TITLE
Fix travis issues for latest keras versions.

### LINKED ISSUE #2865: Investigate and fix Keras problem under Python 3.8
One of our unit tests fails on Travis under Py3.8.

```
=================================== FAILURES ===================================

_____________ TestKerasWord2VecWrapper.testEmbeddingLayerCosineSim _____________

self = <gensim.test.test_keras_integration.TestKerasWord2VecWrapper testMethod=testEmbeddingLayerCosineSim>

    def testEmbeddingLayerCosineSim(self):

        """

        Test Keras 'Embedding' layer returned by 'get_embedding_layer' function for a simple word similarity task.

        """

        keras_w2v_model = self.model_cos_sim

        keras_w2v_model_wv = keras_w2v_model.wv

    

        embedding_layer = keras_w2v_model_wv.get_keras_embedding()

    

        input_a = Input(shape=(1,), dtype='int32', name='input_a')

        input_b = Input(shape=(1,), dtype='int32', name='input_b')

        embedding_a = embedding_layer(input_a)

        embedding_b = embedding_layer(input_b)

        similarity = dot([embedding_a, embedding_b], axes=2, normalize=True)

    

>       model = Model(input=[input_a, input_b], output=similarity)

embedding_a = <tf.Tensor 'embedding_4/Identity:0' shape=(None, 1, 100) dtype=float32>

embedding_b = <tf.Tensor 'embedding_4_1/Identity:0' shape=(None, 1, 100) dtype=float32>

embedding_layer = <tensorflow.python.keras.layers.embeddings.Embedding object at 0x7f603df0d130>

input_a    = <tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>

input_b    = <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>

keras_w2v_model = <gensim.models.word2vec.Word2Vec object at 0x7f603df0d760>

keras_w2v_model_wv = <gensim.models.keyedvectors.Word2VecKeyedVectors object at 0x7f603df0d250>

self       = <gensim.test.test_keras_integration.TestKerasWord2VecWrapper testMethod=testEmbeddingLayerCosineSim>

similarity = <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>

gensim/test/test_keras_integration.py:62: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.tox/py38-linux/lib/python3.8/site-packages/tensorflow/python/keras/engine/training.py:167: in __init__

    super(Model, self).__init__(*args, **kwargs)

        __class__  = <class 'tensorflow.python.keras.engine.training.Model'>

        args       = ()

        kwargs     = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}

        self       = <tensorflow.python.keras.engine.training.Model object at 0x7f603df18ac0>

.tox/py38-linux/lib/python3.8/site-packages/tensorflow/python/keras/engine/network.py:176: in __init__

    self._init_subclassed_network(**kwargs)

        args       = ()

        kwargs     = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}

        self       = <tensorflow.python.keras.engine.training.Model object at 0x7f603df18ac0>

.tox/py38-linux/lib/python3.8/site-packages/tensorflow/python/training/tracking/base.py:456: in _method_wrapper

    result = method(self, *args, **kwargs)

        args       = ()

        kwargs     = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}

        method     = <function Network._init_subclassed_network at 0x7f60465f6d30>

        previous_value = True

        self       = <tensorflow.python.keras.engine.training.Model object at 0x7f603df18ac0>

.tox/py38-linux/lib/python3.8/site-packages/tensorflow/python/keras/engine/network.py:367: in _init_subclassed_network

    self._base_init(name=name, **kwargs)

        kwargs     = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}

        name       = None

        self       = <tensorflow.python.keras.engine.training.Model object at 0x7f603df18ac0>

.tox/py38-linux/lib/python3.8/site-packages/tensorflow/python/training/tracking/base.py:456: in _method_wrapper

    result = method(self, *args, **kwargs)

        args       = ()

        kwargs     = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'name': None, 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}

        method     = <function Network._base_init at 0x7f60465f6a60>

        previous_value = False

        self       = <tensorflow.python.keras.engine.training.Model object at 0x7f603df18ac0>

.tox/py38-linux/lib/python3.8/site-packages/tensorflow/python/keras/engine/network.py:202: in _base_init

    generic_utils.validate_kwargs(kwargs, {'trainable', 'dtype', 'dynamic',

        __class__  = <class 'tensorflow.python.keras.engine.network.Network'>

        kwargs     = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}

        name       = None

        self       = <tensorflow.python.keras.engine.training.Model object at 0x7f603df18ac0>

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

kwargs = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}

allowed_kwargs = {'autocast', 'dtype', 'dynamic', 'trainable'}

error_message = 'Keyword argument not understood:'

    def validate_kwargs(kwargs,

                        allowed_kwargs,

                        error_message='Keyword argument not understood:'):

      """Checks that all keyword arguments are in the set of allowed keys."""

      for kwarg in kwargs:

        if kwarg not in allowed_kwargs:

>         raise TypeError(error_message, kwarg)

E         TypeError: ('Keyword argument not understood:', 'input')

allowed_kwargs = {'autocast', 'dtype', 'dynamic', 'trainable'}

error_message = 'Keyword argument not understood:'

kwarg      = 'input'

kwargs     = {'input': [<tf.Tensor 'input_a_3:0' shape=(None, 1) dtype=int32>, <tf.Tensor 'input_b_3:0' shape=(None, 1) dtype=int32>], 'output': <tf.Tensor 'dot_3/Identity:0' shape=(None, 1, 1) dtype=float32>}
```

### PR DESCRIPTION
**Pull request by dsandeep0138**
Fixes #2865

### CODE REVIEW

**Comment by mpenkov** [comment]
@dsandeep0138 Thank you for the PR. As you can see Travis (and CircleCI, and Azure) have started testing your PR for you. It's automatic.

**Comment by dsandeep0138** [comment]
Okay. Then I may have to unpin keras version, that was pinned to 2.3.1 right? To see if the issue pops up.

**Comment by dsandeep0138** [comment]
Hi @mpenkov , looks like all the 3 tests in `test_keras_integration.py` are passing. Please review.

https://travis-ci.org/github/RaRe-Technologies/gensim/jobs/703058074
``
gensim/test/test_keras_integration.py ...                                [ 37%]
``

File: gensim/test/test_keras_integration.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def testEmbeddingLayerCosineSim(self):
         embedding_b = embedding_layer(input_b)
         similarity = dot([embedding_a, embedding_b], axes=2, normalize=True)
 
-        model = Model(input=[input_a, input_b], output=similarity)
+        model = Model([input_a, input_b], similarity)
```
Comment (piskvorky) [inline gensim/test/test_keras_integration.py:62]: Named parameters are both safer and more descriptive. Are the new plural names (Keras 2.4.0+) backward compatible with the older 2.3.1?

If so let's use those.

Comment (dsandeep0138) [reply gensim/test/test_keras_integration.py:62]: I just verified on keras 2.3.1 that both "input" and "inputs" work. Let me raise a new patch using named parameters inputs, and outputs.

For verification, I will make the changes in two-phases, one with keras restricted to 2.3.1 and other without any keras version pin (so, keras will be 2.4.3) to make sure it works for both.

**Comment by dsandeep0138** [comment]
Alright! Named params "inputs" and "outputs" work on keras 2.3.1, I'll unpin the keras version now for one final run.

**Comment by dsandeep0138** [comment]
@piskvorky @gojomo @mpenkov Please review.

**Comment by gojomo** [comment]
My superficial review would be: 

Not being familiar with either the relevant keras API, or our 'keras integration', this appears to be a tiny syntactic update, and it makes all tests pass, so sure, merge it. 

But looking more closely at both the tested method & test code:

This utility function doesn't really belong cluttering up the `KeyedVectors` class: it's a fairly trivial bit of data-translation, very specific to some narrow uses, with an extra package requirement. It could & should be a in a keras-specific submodule that's only imported by people using keras, and could take the `KeyedVectors` instance as an argument (rather than `self`). 

Or, in the spirit of #2852 ("reduce gensim surface area"), it could be moved to a separate package (`gensim-extras`) - so it doesn't encumber testing/development of core gensim, or removed in favor of a tiny bit of example code, "here's how you convert gensim KeyedVectors into a keras Embedding", in docs, an example notebook, a 'recipes' webpage, or a blog post. It's 2-6 lines of trifling code, but its test failure due to a 'cosmic ray' from another project has soaked up a bunch of attention/time from 4 people who could be making other more important gensim improvements. 

Looking at the `test_keras_integration.py` code, it also failed because it was insufficiently focused on the gensim functionality.  It's not checking if the return value is just a translation of gensim values into a legal/sensible `Embedding` - it's doing some full-cycle `Word2Vec` training and some full-cycle `keras` training. It's one of the 20 slowest tests in our whole test suite – sure, only ~9 seconds, but of negligible value usually over the many hundreds/thousands of times it's triggered as a result of other changes, and of negative value in this case. 

(Gensim has a bunch of tests that take a lot of time but don't test much real, just completion-without-error, and often which deeply test inner structural details rather than meaningful results, and thus serve as a drag on sensible refactorings when they trigger test failures that require maintenance without tangible benefits to developer or gensim end-users.)

To advance #2852, I'd copy the existing method code to a new wiki page on "using gensim with keras", leave a stub pointing users there, and remove the functionality/tests from gensim so there will be no further maintenance cost.

**Pull request merged**